tests/delete_regress/tests.py

```diff
@@ -255,9 +255,9 @@ def test_delete_proxy_pair(self):
         self.assertEqual(len(FooFileProxy.objects.all()), 0)
 
     def test_19187_values(self):
-        with self.assertRaises(TypeError):
+        with self.assertRaises(AssertionError):
```

Comment:
Vague possibility of backwards compatibility issues also (in user code which modifies an arbitrary qs passed in and catches the `TypeError`). I would like to see the other errors changed given the option.